tests/psycopg2_test.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+import tempfile
+import shutil
+
+import psycopg2
+
+from pg_temp import TempDB, init_temp_db
+
+
+class Psycopg2TestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.db = TempDB(databases=['test_db'], verbosity=1)
+
+    def tearDown(self):
+        self.db.cleanup()
+
+    def test_db_connection(self):
+        connection = psycopg2.connect(host=self.db.pg_socket_dir,
+                                      database='test_db')
+        self.assertTrue(connection)
+        connection.close()
+
+
+class InitTempDBTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.db = init_temp_db(databases=['test_db'], verbosity=1)
+
+    def tearDown(self):
+        self.db.cleanup()
+
+    def test_db_connection(self):
+        connection = psycopg2.connect(host=self.db.pg_socket_dir,
+                                      database='test_db')
+        self.assertTrue(connection)
+        connection.close()
+
+
+class SpecifySocketTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.socket_dir = tempfile.mkdtemp()
+        self.db = TempDB(databases=['test_db'], verbosity=1,
+                         sock_dir=self.socket_dir)
+
+    def tearDown(self):
+        self.db.cleanup()
+        shutil.rmtree(self.socket_dir)
+
+    def test_db_connection(self):
+        connection = psycopg2.connect(host=self.db.pg_socket_dir,
+                                      database='test_db')
+        self.assertTrue(connection)
+        connection.close()
```

Comment:
Nicely done!  This reads so straightforward and correctly to me.

One sug while I've got you here ;-)

Since we're using pytest now, we could choose to switch over to the pytest-friendly `assert connection` assertion style instead of `self.assertTrue(connection)`.  And since pytest allows us to write top-level functions instead of requiring all the class boilerplate, this test could be boiled down to these bits:

```python
def test_socket_dir():
    socket_dir = tempfile.mkdtemp()
    db = TempDB(databases=['test_db'], verbosity=1, sock_dir=socket_dir)
    connection = psycopg2.connect(host=db.pg_socket_dir, database='test_db')
    assert connection
    connection.close()
    db.cleanup()
    shutil.rmtree(socket_dir)
```

For the `__main__` bit, the `pytest` equivalent when running the file directly is...

```python
if __name__ == '__main__':
    pytest.main([__file__])
```

I've been using that pattern since it allows suites where use of `unittest` is optional.  Without the `__main__` bit it won't run the top-level `test_foo()` functions when you run `python3 tests/test_foo.py`.

Reply:
Ok, I think I'll simplify the test file in a future commit (and maybe push an 0.8.1). For now I'll cut a release since I want to use this stuff :)